drivers/dma/dma_mcux_lpc.c

```diff
@@ -298,14 +297,8 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 		return -EINVAL;
 	}
 
-#if defined FSL_FEATURE_DMA_NUMBER_OF_CHANNELS
```

Comment:
Doesn't total DMA channels here mean something different than `dev_config->num_of_channels`? If I understand correctly:
- `dev_config->num_of_channels`: the number of channel slots we expect the driver to allocate (can be less than number of channels in HW)
- `total_dma_channels`: the number of channels that are actually present on the hardware.

Reply:
`num-of-channels` sets the number of channel configuration structures that the driver allocates. You can see here: https://github.com/nxp-zephyr/zephyr/blob/fix/remove_lpc_dma_macro/drivers/dma/dma_mcux_lpc.c#L349 that we allocate a "virtual channel" value when a new channel is configured. With this PR, you are effectively requiring that the `number-of-channels` property be equivalent to the number of channels the hardware supports.

Based off the documentation of `number-of-channels`, it looks like this driver was previously misusing this DT property. Given that, I think a change here makes sense. I see two ways we could fix this:

- require that the DMA driver statically allocate `num-of-channels` DMA configuration structures, and remove the logic to handle virtual channels from the driver (so the channel number always matches the index of the channel configuration structure)
- retain the ability to configure the number of statically allocated DMA config structures, but make this property configurable via a Kconfig or another nxp specific devicetree property. This way, we do not misuse this `number-of-channels` devicetree property anymore.

Personally, I would prefer the second approach, but I am open to feedback.

Reply:
Or, if it is ever actually needed to pretend there are less dma channels in hardware than there actually are, one can just reduce the number of channels DT property, and everything will work the same way. I think you are over-complicating this.

Reply:
Yes, there is an argument for a Kconfig that supports a case where a user might want to use slightly less RAM by allocating less channels than needed, but this is a functionality that in my opinion we don't correctly support yet, so it is outside the scope of the PR. 

The devicetree should describe hardware not software configuration, and that's what this PR is fixing, as well as removing the reliance on SDK macros which are notoriously unstable.

Reply:
this should be the count of hardware channels imo, so this is the right approach now

Reply:
@teburd, if you are okay with this PR can you give it an approval?